elastic/datadog_checks/elastic/elastic.py

```diff
@@ -48,79 +48,79 @@ def __init__(self, name, init_config, instances):
                 'name': 'aws_host',
                 'default': urlparse(self.instance['url']).hostname,
             }
+        self.config = from_instance(self.instance)
 
-    def check(self, instance):
-        config = from_instance(instance)
-        admin_forwarder = config.admin_forwarder
-        jvm_rate = instance.get('gc_collectors_as_rate', False)
+    def check(self, _):
+        admin_forwarder = self.config.admin_forwarder
+        jvm_rate = self.instance.get('gc_collectors_as_rate', False)
 
         # Check ES version for this instance and define parameters
         # (URLs and metrics) accordingly
         try:
-            version = self._get_es_version(config)
+            version = self._get_es_version()
         except AuthenticationError:
             self.log.exception("The ElasticSearch credentials are incorrect")
             raise
 
-        health_url, stats_url, pshard_stats_url, pending_tasks_url = self._get_urls(version, config.cluster_stats)
+        health_url, stats_url, pshard_stats_url, pending_tasks_url = self._get_urls(version, self.config.cluster_stats)
         stats_metrics = stats_for_version(version, jvm_rate)
-        if config.cluster_stats:
+        if self.config.cluster_stats:
             # Include Node System metrics
             stats_metrics.update(node_system_stats_for_version(version))
         pshard_stats_metrics = pshard_stats_for_version(version)
 
         # Load stats data.
         # This must happen before other URL processing as the cluster name
         # is retrieved here, and added to the tag list.
-        stats_url = self._join_url(config.url, stats_url, admin_forwarder)
-        stats_data = self._get_data(stats_url, config)
+        stats_url = self._join_url(self.config.url, stats_url, admin_forwarder)
+        stats_data = self._get_data(stats_url)
```

Comment:
Small nit but just wondering why you got rid of the `config` parameter in `_get_data()` but kept `self.config.url` in `_join_url()`?

Reply:
No reason in particular, will remove it 👍